### PR TITLE
feat(list): Optional output as JSON and YAML

### PR DESCRIPTION
**Pull request by seaneagan**
The choice of interface `--output (json|yaml)` is to match that of the
status command, except that -o is not available as it is already used
by --offset.

WIP #1534

### CODE REVIEW

**Comment by seaneagan** [comment]
@jascott1 is anyone able to take a look at this one?

File: cmd/helm/list.go [hunk 1]
```diff
@@ -233,23 +246,98 @@ func (l *listCmd) statusCodes() []release.Status_Code {
...
+	var finalResult interface{}
+	if short {
+		shortResult = shortenListResult(result)
+		finalResult = shortResult
+	} else {
+		finalResult = result
+	}
+
+	switch format {
+	case "":
```
Comment (bacongobbler) [inline cmd/helm/list.go]: We should probably accept `--format=plain` as an available option instead, and make that the default. `--format=""` seems a little odd from a UX perspective on its intended behaviour. What do you think?

Comment (bacongobbler) [reply cmd/helm/list.go]: Actually it looks like this is consistent with `helm status --output=""`, so we already have precedent here.

**Review by bacongobbler — APPROVED** [review]
manually tested, LGTM

**Comment by bacongobbler** [comment]
needs sign off from another maintainer though since it's a `size/L`

**Review by jascott1 — COMMENTED** [review]
LGTM

**Comment by jwatte** [comment]
Which version did this merge into?

    [12:55:08] jwatte@xps15:~/observe$ helm version
    Client: &version.Version{SemVer:"v2.9.1", GitCommit:"20adb27c7c5868466912eebdf6664e7390ebe710", GitTreeState:"clean"}
    Server: &version.Version{SemVer:"v2.9.1", GitCommit:"20adb27c7c5868466912eebdf6664e7390ebe710", GitTreeState:"clean"}

    [12:55:11] jwatte@xps15:~/observe$ helm list --format=text
    Error: unknown flag: --format

**Comment by bacongobbler** [comment]
v2.10.0.

**Comment by eiva** [comment]
Still nothing...
```
> helm list --format=json
Error: unknown flag: --format
> helm version
Client: &version.Version{SemVer:"v2.14.1", GitCommit:"5270352a09c7e8b6e8c9593002a73535276507c0", GitTreeState:"clean"}
```

**Comment by bacongobbler** [comment]
Use `helm list --output json`.

**Comment by iftachsc** [comment]
i have helm client v2.12.1 how come it doesnt have --format param?

**Comment by bacongobbler** [comment]
In helm 2, it’s `helm list —output json`. See above^

**Comment by iftachsc** [comment]
i am getting
Error: unknown flag: --output

**Comment by iftachsc** [comment]
i was looking for this for helm search comand. not ls.
i am getting Error: unknown flag: --output
anyway i solved it with curl call to the api - it returns json

**Comment by magmax** [comment]
what about `template`?
```
$ helm version                                                                                                                                                                                                                                                                        
version.BuildInfo{Version:"v3.5.4", GitCommit:"1b5edb69df3d3a08df77c9902dc17af864ff05d1", GitTreeState:"clean", GoVersion:"go1.15.11"}
$ helm template --output json .
Error: unknown flag: --output
$ helm template --format json .                                                                                                                                                                                                                                                       
Error: unknown flag: --format
$
```

**Pull request merged**